File: composition/java/Shapes.java
public class Shapes {
  public static void main(String[] args) {
    Rectangular rectangle = new Rectangle(4, 5, 0, 0);
    System.out.println(rectangle.toString() + "\n");
    rectangle.rotate(); System.out.println("rotate");
    System.out.println(rectangle.toString() + "\n");
    rectangle.augmentWidthBy(2); System.out.println("augment width by 2");
    System.out.println(rectangle.toString() + "\n");

    System.out.println("########################################################################");

    Square rectangle2 = new Square(4, 0, 0);
    System.out.println(rectangle2.toString() + "\n");
    rectangle2.rotate(); System.out.println("rotate");
    System.out.println(rectangle2.toString() + "\n");
    Square square = (Square) rectangle2.augmentSideBy(2); System.out.println("augment side by 2");
    System.out.println(square.toString() + "\n");
  }
}

File: composition/ruby/shapes.rb
#!/usr/bin/env ruby
require 'rectangle'
require 'square'

rectangle = Rectangle.new(4, 5, 0, 0)
puts rectangle.to_s, "\n"
rectangle.rotate!; puts "rotate"
puts rectangle.to_s, "\n"
rectangle.augment_width_by!(2); puts "augment width by 2"
puts rectangle.to_s, "\n"

puts "###########################################################"

square = Square.new(4, 0, 0)
puts square.to_s, "\n"
square.rotate!; puts "rotate"
puts square.to_s, "\n"
square.augment_side_by!(2); puts "augment side by 2"
puts square.to_s, "\n"